src/main/java/com/erigitic/jobs/TEAction.java

```diff
@@ -0,0 +1,196 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.erigitic.jobs;
+
+import com.google.inject.Inject;
+import ninja.leaping.configurate.ConfigurationNode;
+import org.slf4j.Logger;
+import org.spongepowered.api.block.BlockState;
+import org.spongepowered.api.block.trait.BlockTrait;
+import org.spongepowered.api.data.DataQuery;
+
+import java.util.*;
+
+public class TEAction {
+
+    public static TEAction fromActionNode(String action, ConfigurationNode node) {
+        ConfigurationNode growthTraitNode = node.getNode("growthTrait");
+        if (growthTraitNode == null)
+            growthTraitNode = node.getNode("growth-trait");
+        ConfigurationNode idTraitNode = node.getNode("id-trait");
+        if (idTraitNode.isVirtual()) {
+            return new TEAction(action, node.getKey().toString(), TEActionReward.getFromNode(node), growthTraitNode.getString(null));
+        }
+        String traitName = idTraitNode.getString("type");
+        if (traitName == null)
+            return new TEAction();
+        ConfigurationNode idTraits = node.getNode(traitName);
+        Map<String, TEActionReward> rewardMap = new HashMap<>();
+        if (idTraits.hasMapChildren()) {
+            idTraits.getChildrenMap().forEach((k, v) -> {
+                if (!(k instanceof String))
+                    return;
+                TEActionReward reward = TEActionReward.getFromNode(v);
+                rewardMap.put(((String) k), reward);
+            });
+        }
+        return new TEAction(action, node.getKey().toString(), rewardMap, idTraitNode.getString("type"), growthTraitNode.getString(null));
+    }
+
+    private String action;
+    private String targetID;
+
+    /* Complex stuff if we need to evaluate traits */
+    private String growthTrait = null;
+    private String idTrait = null;
+    private Map<String, TEActionReward> rewards;
+
+    /* Simple stuff where we don't */
+    private TEActionReward reward;
+
+    private boolean isValid = false;
+
+    // Just construct an invalid one
+    protected TEAction() {}
+
+    public TEAction(String action, String targetID, TEActionReward reward, String growthTrait) {
+        this.action = action;
+        this.targetID = targetID;
+        this.growthTrait = growthTrait;
+        this.reward = reward;
+        isValid = (action != null && !action.isEmpty() && targetID != null && !targetID.isEmpty());
+    }
+
+    public TEAction(String action, String target, Map<String, TEActionReward> rewards, String idTrait, String growthTrait) {
+        this(action, target, null, growthTrait);
+        this.idTrait = idTrait;
+        this.rewards = rewards;
+    }
+
+    public boolean isGrowing() { return growthTrait != null; }
```

Comment:
Avoid single line functions for consistency

This should be done throughout this and the rest of the changed files.